src/Microsoft.AspNetCore.JsonPatch/Helpers/ObjectTreeAnalyisResult.cs

```diff
@@ -127,12 +138,15 @@ internal class ObjectTreeAnalysisResult
             // case, it's valid for add if there's 1 item left in the propertyPathTree.
             //
             // it can also be a property info.  In that case, if there's nothing left in the path
-            // tree we're at the end, if there's one left we can try and set that. 
-            if (targetObject is IDictionary<string, object>)
+            // tree we're at the end, if there's one left we can try and set that.
+            var idictionaryType = ClosedGenericMatcher.ExtractGenericInterface(
```

Comment:
Looks like `ReferenceEquals(targetObject, dictionaryWrapper)` (w/ new name for `dictionaryWrapper`) would work fine for the condition below. No need for another `ClosedGenericMatcher.ExtractGenericInterface()` call.